e2e/cypress/integration/notifications/ignore_channel_mentions.js

```diff
@@ -0,0 +1,138 @@
+}
+
+function setIgnoreMentions(toSet) {
+    let stringToSet = 'Off';
+    if (toSet) {
+        stringToSet = 'On';
+    }
+
+    // # Click on the header to open dropdown
+    cy.get('#channelHeaderDropdownButton').should('exist').click();
+
+    // # Click on the notification preferences
+    cy.get('#channelNotificationPreferences').should('exist').click();
+
+    // # Click on the edit button for ignore channel mentions
+    cy.get('#ignoreChannelMentionsEdit').should('exist').click();
+
+    // # Click on selected option
+    cy.get(`#ignoreChannelMentions${stringToSet}`).should('exist').click();
+
+    // # Click on save to save the configuration
+    cy.get('#saveSetting').should('exist').click();
+
+    // * Assert that the option selected is reflected
+    cy.get('#ignoreChannelMentionsDesc').should('contain', stringToSet);
+
+    // # Click on the X button to close the modal
+    cy.get('#channelNotificationModalLabel').siblings('.close').click();
+}
+
+describe('CS15445 Join/leave messages', () => {
+    let testTeam;
+    let userB;
+
+    let channelA;
+    let channelB;
+
+    before(() => {
+        // # Login as new user and visit town-square
+        cy.apiInitSetup().then(({team, user}) => {
+            testTeam = team;
+
+            // # Add 1 user
+            cy.apiCreateUser().then(({user: newUser}) => {
+                userB = newUser;
+                cy.apiAddUserToTeam(testTeam.id, userB.id);
+            });
+
+            // # Create two channels
+            cy.apiCreateChannel(testTeam.id, 'channel-test', 'Channel').then((res) => {
+                channelA = res.body;
+            });
+            cy.apiCreateChannel(testTeam.id, 'channel-test', 'Channel').then((res) => {
+                channelB = res.body;
+            });
+
+            cy.apiLogin(user);
+        });
+    });
+
+    it('MM-T567 - Turn on Ignore mentions for @channel, @here and @all', () => {
+        cy.visit(`/${testTeam.name}/channels/${channelA.name}`);
+
+        // # Add users to channel
+        addNumberOfUsersToChannel(1);
+
+        cy.getLastPostId().then((id) => {
+            // * The system message should contain 'added to the channel by you'
+            cy.get(`#postMessageText_${id}`).should('contain', 'added to the channel by you');
+        });
+
+        // # Set ignore mentions
+        setIgnoreMentions(true);
+
+        // # Go to a different channel
+        cy.visit(`/${testTeam.name}/channels/${channelB.name}`);
+
+        // # Post messages as another user on the first channel
+        cy.postMessageAs({sender: userB, message: '@all test', channelId: channelA.id});
+        cy.postMessageAs({sender: userB, message: '@channel test', channelId: channelA.id});
+        cy.postMessageAs({sender: userB, message: '@here test', channelId: channelA.id});
+
+        // * Assert the channel is unread with no mentions
+        cy.get(`#sidebarItem_${channelA.name}`).should('have.class', 'unread-title');
```

Comment:
```suggestion
        cy.get(`#sidebarItem_${channelA.name}`).wait(TIMEOUTS.ONE_SEC).should('have.class', 'unread-title');
```

Reply:
Cypress was trying to verify before everything was even loaded, so suggesting to add 1 second timeout.